### ISSUE TITLE
Missing includes in jpegls_error.cpp

### ISSUE THREAD

**Issue by SpaceIm** [team-charls/charls#62]
Clang <= 5 and AppleClang 9.3 fail to compile jpegls_error.cpp if using libc++ (and not libstdc++) because string (and system_error) are not included.

```
[ 44%] Building CXX object source_subfolder/CMakeFiles/charls.dir/src/jpegls_error.cpp.o
/home/conan/.conan/data/charls/2.1.0/bincrafters/testing/build/2b9f4a7d83ec1ec57c27b92740c5dea74386cdf8/source_subfolder/src/jpegls_error.cpp:19:12: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    string message(int error_value) const override
           ^
/usr/include/c++/v1/iosfwd:193:33: note: template is declared here
    class _LIBCPP_TYPE_VIS_ONLY basic_string;
                                ^
/home/conan/.conan/data/charls/2.1.0/bincrafters/testing/build/2b9f4a7d83ec1ec57c27b92740c5dea74386cdf8/source_subfolder/src/jpegls_error.cpp:21:16: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
        return charls_get_error_message(static_cast<jpegls_errc>(error_value));
               ^
/usr/include/c++/v1/iosfwd:193:33: note: template is declared here
    class _LIBCPP_TYPE_VIS_ONLY basic_string;
```

**Comment by vbaderks** [comment]
Thanks for report. Would a fix on "master" be enough?

Note: system_error should be included by means of jpegls_error.h

**Comment by SpaceIm** [comment]
@vbaderks it doesn't matter for me. I only create package of releases for conan, so anyway I have to create a patch for 2.1.0 ;)

**Comment by vbaderks** [comment]
This has been resolved with commit 5a58c40f0fd38a75cbd3b1ac2d9dc6171a832f47.

It can be tested on Ubuntu 16.04 (WSL) with clang 4.0. The CI builds are using newer versions of clang and libc++ and don't have that problem because the system_error header already included the string header.
